packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkImplementation.kt

```diff
@@ -150,26 +166,16 @@ class DdSdkImplementation(
         }
     }
 
-    private fun handlePostFrameCallbackError(e: IllegalStateException) {
-        datadog.telemetryError(e.message ?: MONITOR_JS_ERROR_MESSAGE, e)
-    }
-
-    private fun monitorJsRefreshRate(ddSdkConfiguration: DdSdkConfiguration) {
-        val frameTimeCallback = buildFrameTimeCallback(ddSdkConfiguration)
-        if (frameTimeCallback != null) {
-            reactContext.runOnJSQueueThread {
-                val vitalFrameCallback =
-                    VitalFrameCallback(frameTimeCallback, ::handlePostFrameCallbackError) {
-                        initialized.get()
-                    }
-                try {
-                    Choreographer.getInstance().postFrameCallback(vitalFrameCallback)
-                } catch (e: IllegalStateException) {
-                    // This should never happen as the React Native thread always has a Looper
-                    handlePostFrameCallbackError(e)
-                }
-            }
+    private fun createFrameRateProvider(
+        ddSdkConfiguration: DdSdkConfiguration
+    ): FrameRateProvider? {
+        val frameTimeCallback = buildFrameTimeCallback(ddSdkConfiguration) ?: return null
+        val frameRateProvider = FrameRateProvider(frameTimeCallback)
+        reactContext.runOnJSQueueThread {
+            frameRateProvider.start()
```

Comment:
safety check: is `runOnJSQueueThread` a blocking call (does it wait until the task is actually executed on the main thread) or not? if it is not, then instance can be returned before it is started, not sure if it can cause any issue.

Reply:
I believe it won't cause any problems. The actual callback where the frame-rate is received is invoked by the `frameRateProvider` only after it is started. There should not be any case where the provider is accessed in an illegal state.